### ISSUE TITLE
Implement ProcessPerChunk

### ISSUE THREAD

**Issue by mmomtchev** [mmomtchev/node-gdal-async#3]
`GDALMDArray::ProcessPerChunk` is a very interesting mechanism - that calls for some very interesting solutions in JS.

* A naive, sync, event-blocking solution where GDAL calls a JS callback to process the data - limited to one thread at a time blocking the event loop
* A set of primitives implemented in C++ that can be pushed from JS to be run in parallel C++ in async mode
* A more complex approach allowing the JS code to construct and push a pipeline to be run in parallel C++ in async mode

Anything but the very first solution is probably out of scope for 3.3

**Comment by mmomtchev** [comment]
This will be implemented using https://github.com/mmomtchev/exprtk.js - which allows for the most complex pipelines possible
